treeherder/intermittents_commenter/commenter.py

```diff
@@ -63,41 +63,43 @@ def generate_bug_changes(self, startday, endday, alt_startday, alt_endday):
         for bug_id, counts in iteritems(bug_stats):
             change_priority = None
             bug_info = None
-            whiteboard = None
+            change_whiteboard = None
             priority = 0
             rank = None
 
-            # recommend disabling when more than 150 failures tracked over 21 days
-            if alt_bug_stats[bug_id]['total'] >= 150:
-                bug_info, whiteboard = self.check_bug_info(bug_info, bug_id)
-                # if fetch_bug_details fails (called in check_bug_info), None is returned
-
-                if whiteboard is not None and not self.check_whiteboard_status(whiteboard):
-                    priority = 3
-                    whiteboard = self.update_whiteboard(whiteboard, '[stockwell disable-recommended]')
-
             if self.weekly_mode:
                 priority = self.assign_priority(priority, counts)
                 if priority == 2:
-                    bug_info, whiteboard = self.check_bug_info(bug_info, bug_id)
-                    # if fetch_bug_details fails (called in check_bug_info), None is returned
-                    if bug_info is not None:
-                        change_priority, whiteboard = self.check_needswork_owner(change_priority, bug_info, whiteboard)
+                    bug_info = self.fetch_bug_details(TRIAGE_PARAMS, bug_id) if not bug_info else bug_info
+                    # if fetch_bug_details fails, None is returned
+
+                    if bug_info:
+                        change_priority, change_whiteboard = self.check_needswork_owner(change_priority, bug_info)
 
                 # change [stockwell needswork] to [stockwell unknown] when failures drop below 20 failures/week
                 if (counts['total'] < 20):
-                    bug_info, whiteboard = self.check_bug_info(bug_info, bug_id)
-                    # if fetch_bug_details fails (called in check_bug_info), None is returned
-                    if whiteboard is not None:
-                        whiteboard = self.check_needswork(whiteboard)
+                    bug_info = self.fetch_bug_details(TRIAGE_PARAMS, bug_id) if not bug_info else bug_info
+                    # if fetch_bug_details fails, None is returned
+
+                    if bug_info:
+                        change_whiteboard = self.check_needswork(change_whiteboard or bug_info['whiteboard'])
```

Comment:
This will override line 77 if change_whiteboard == [needswork:owner]. It looks like it's doing that in the old OF code, but please confirm you want this to happen.

Reply:
Yes, that's right. It doesn't matter much, since priority == 2 implies counts['total'] > 30: this block and the previous one are mutually exclusive.

Reply:
oh right :) I'll nix passing change_whiteboard to check_needswork for clarity.